api/controllers/StartController.cs

```diff
@@ -46,78 +42,79 @@ public async Task<StatusChangeOutputModel> StartServicesAsync(InputRequestModel
                 string liveOutputName = $"LiveStreamingApi-LiveOutput-{liveEventName}-{Guid.NewGuid()}";
                 string streamingLocatorName = $"LiveStreamingApi-StreamingLocator-{liveEventName}-{Guid.NewGuid()}";
 
-                // 3. Abort if the current live event is not stopped
-                StatusOutputModel.Resource liveEvent = preRunServiceStatus
-                    .LiveEvents
-                    .FindLast(currentEvent => currentEvent.Name == liveEventName);
-
-                if (liveEvent.Status != ResourceStatusEnum.Stopped)
+                if (!IsLiveEventStopped(preRunServiceStatus, liveEventName))
                 {
                     continue;
                 }
 
-                // 4. Create the asset
-                Asset asset = await Client.Assets.CreateOrUpdateAsync(
-                    resourceGroupName: Config.ResourceGroup,
-                    accountName: Config.AccountName,
-                    assetName: assetName,
-                    parameters: new Asset()
-                );
-
-                // 5. Create the Live Output
-                LiveOutput liveOutput = new LiveOutput(
-                    assetName: asset.Name,
-                    manifestName: manifestName,
-                    archiveWindowLength: TimeSpan.FromMinutes(10)
-                );
-
-                await Client.LiveOutputs.CreateAsync(
-                    resourceGroupName: Config.ResourceGroup,
-                    accountName: Config.AccountName,
-                    liveEventName: liveEventName,
-                    liveOutputName: liveOutputName,
-                    parameters: liveOutput
-                );
+                Asset asset = await CreateAsset(assetName);
+                await CreateLiveOutput(asset, liveEventName, liveOutputName, manifestName);
+                await CreateStreamingLocator(asset, streamingLocatorName);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/literal-life-church/live-streaming-api/pullRequest?prid=5532272)